src/fancy-tools/magnifyTool.js

```diff
@@ -0,0 +1,253 @@
+/* eslint no-underscore-dangle: 0 */
+/* eslint class-methods-use-this: 0 */
+import external from '../externalModules.js';
+import { getBrowserInfo } from '../util/getMaxSimultaneousRequests.js';
+import { clipToBox } from '../util/clip.js';
+import { getNewContext, fillBox } from '../util/drawing.js';
+import baseTool from './../base/baseTool.js';
+
+const cornerstone = external.cornerstone;
+
+export default class extends baseTool {
+  constructor (name = 'magnify') {
+    super({
+      name,
+      supportedInteractionTypes: ['mouse', 'touch'],
+      configuration: {
+        magnifySize: 300,
+        magnificationLevel: 2
+      }
+    });
+
+    this.browserName = undefined;
+    this.zoomCanvas = undefined;
+    this.zoomElement = undefined;
+
+    // Needed for Safari Canvas Hack
+    if (!this.browserName) {
+      const infoString = getBrowserInfo();
+      const info = infoString.split(' ');
+
+      this.browserName = info[0];
+    }
+
+    // Mode Callbacks: (element, options)
+    this.activeCallback = this._createMagnificationCanvas.bind(this);
+    this.enableCallback = this._createMagnificationCanvas.bind(this);
+    this.disableCallback = this._destroyMagnificationCanvas.bind(this);
+
+    // Touch
+    this.postTouchStartCallback = this._addMagnifyingGlass.bind(this);
```

Comment:
Going back to the mouse && touch support discussion, I quite like how you are dealing with the interfacing here. If this approach is useful for most/all tools, I'm leaning toward the one class for both interaction types, with implementation in the class body and the interfacing in the constructor. Good job 👍 .